src/org/labkey/test/components/ui/entities/EntityInsertPanel.java

```diff
@@ -204,19 +330,45 @@ protected ElementCache newElementCache()
 
     protected class ElementCache extends Component.ElementCache
     {
+        public ElementCache()
+        {
+            waitForLoaded();
+        }
+
+        Locator modeSelectListItem(String containsText)
+        {
+            return Locator.tagWithClass("li", "list-group-item").containing(containsText);
+        }
+
         Locator deleteRowsBtnLoc = Locator.XPathLocator.union(
                 Locator.button("Delete rows"),
                 Locator.buttonContainingText("Remove"));
+        Locator bulkInsertBtnLoc = Locator.button("Bulk Insert");
+
+        WebElement bulkInsertBtn()
+        {
+            return bulkInsertBtnLoc.findElement(this);
+        }
+        WebElement bulkUpdateBtn()
+        {
+            return Locator.button("Bulk update").findElement(this);
+        }
+        WebElement deleteRowsBtn()
+        {
+            return deleteRowsBtnLoc.findElement(this);
+        }
```

Comment:
Is switching these buttons to getters necessary? The element cache gets cleared when switching modes.

Reply:
Probably not.  I'll switch them back to cached lazyWebElements, with short searchTimeouts